src/vs/editor/contrib/codeAction/browser/codeActionModel.ts

```diff
@@ -197,12 +205,82 @@ export class CodeActionModel extends Disposable {
 					return;
 				}
 
-				const actions = createCancelablePromise(token => getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token));
+				const startPosition = trigger.selection.getStartPosition();
+
+				const actions = createCancelablePromise(async token => {
+					if (this._settingEnabledNearbyQuickfixes() && trigger.trigger.type === CodeActionTriggerType.Invoke && trigger.trigger.triggerAction === CodeActionTriggerSource.QuickFix) {
+						const codeActionSet = await getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token);
+
+						if (token.isCancellationRequested) {
+							return emptyCodeActionSet;
+						}
+
+						// Search for quickfixes in the curret code action set.
+						const foundQuickfix = codeActionSet.validActions?.some(action => action.action.kind ? CodeActionKind.QuickFix.contains(new CodeActionKind(action.action.kind)) : CodeActionKind.None);
+
+						if (!foundQuickfix) {
+							const allMarkers = this._markerService.read({ resource: model.uri });
+
+							// If markers exists, and there are no quickfixes found or length is zero, check for quickfixes on that line.
+							if (allMarkers.length > 0) {
+								const currPosition: Position = trigger.selection.getPosition();
+								let trackedPosition: Position = currPosition;
+								let distance = Number.MAX_VALUE;
+								let toBeModified = false;
+
+								allMarkers.forEach((marker: IMarker) => {
+									const col = marker.endColumn;
+									const row = marker.endLineNumber;
+									const startRow = marker.startLineNumber;
+
+									// Found quickfix on the same line and check relative distance to other markers
+									if ((row === currPosition.lineNumber || startRow === currPosition.lineNumber) && Math.abs(currPosition.column - col) < distance) {
+										distance = Math.abs(currPosition.column - col);
+										toBeModified = true;
+										trackedPosition = new Position(row, col);
+									}
+								});
+
+								// Only retriggers if actually found quickfix on the same line as cursor
+								if (toBeModified) {
+
+									const newCodeActionTrigger: CodeActionTrigger = {
+										type: trigger.trigger.type,
+										triggerAction: trigger.trigger.triggerAction,
+										filter: { include: CodeActionKind.QuickFix },
```

Comment:
If the user requested a specific type of quick fix in the original `trigger.filter`, we should include that instead of the generic `QuickFix`